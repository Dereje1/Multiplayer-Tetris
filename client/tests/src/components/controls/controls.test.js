import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Controls from '../../../../src/components/controls/controls'
import {gameStub} from '../../../stub'

describe('The Controls', () => {
    let props;
    beforeEach(() => {
        props = {
            onhandlePause: jest.fn(),
            onMultiPlayer: jest.fn(),
            onFloorRaise: jest.fn(),
            onReset: jest.fn(),
            minorCanvas:{},
            game:gameStub,
            multiPlayer:[],
            difficulty: 3,
            socketId: 'stub_socket_Id',
            pauseButtonState: false,
            allowMultiPlayer: false,
        }
    })
    afterEach(() => {
        props = null
    })
    test('Will render the controls for single player mode', () => {
        const wrapper = shallow(<Controls  {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('Will render the controls for single player mode with an option to switch to multiplayer mode', () => {
        const updatedProps = {...props, allowMultiPlayer: true}
        const wrapper = shallow(<Controls  {...updatedProps} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test('Will render the controls for multiplayer player mode', () => {
        const updatedProps = {...props, multiPlayer:[true, false]}
        const wrapper = shallow(<Controls  {...updatedProps} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})