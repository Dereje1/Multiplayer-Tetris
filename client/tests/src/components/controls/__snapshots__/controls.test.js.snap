// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Controls Will render the controls for multiplayer player mode 1`] = `
<div
  className="controls"
>
  <canvas
    height={150}
    tabIndex="0"
    width={210}
  />
  <div
    className="controls__resetPause"
  >
    <PlayControls
      disabled={true}
      onReset={[MockFunction]}
      onhandlePause={[MockFunction]}
      pauseButtonState={false}
    />
  </div>
  <FontAwesomeIcon
    beat={false}
    beatFade={false}
    border={false}
    bounce={false}
    className="controls__multiplayer"
    fade={false}
    fixedWidth={false}
    flip={false}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f007",
          "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
        ],
        "iconName": "user",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    maskId={null}
    onClick={[MockFunction]}
    pull={null}
    pulse={false}
    rotation={null}
    shake={false}
    size={null}
    spin={false}
    spinPulse={false}
    spinReverse={false}
    swapOpacity={false}
    symbol={false}
    title=""
    titleId={null}
    transform={null}
  />
  <div
    style={
      Object {
        "height": 40,
      }
    }
  />
  <span>
    Lines Cleared
  </span>
  <span>
    0
  </span>
  <span>
    Difficulty
  </span>
  <span>
    3
  </span>
  <span>
    Multi Player Mode
  </span>
  <span
    className="controls__socket"
  >
    stub_socket_Id
  </span>
</div>
`;

exports[`The Controls Will render the controls for single player mode 1`] = `
<div
  className="controls"
>
  <canvas
    height={150}
    tabIndex="0"
    width={210}
  />
  <div
    className="controls__resetPause"
  >
    <PlayControls
      onReset={[MockFunction]}
      onhandlePause={[MockFunction]}
      pauseButtonState={false}
    />
  </div>
  <ForwardRef(Button)
    className="controls__raise"
    onClick={[MockFunction]}
    variant="contained"
  >
    Raise Floor
  </ForwardRef(Button)>
  <span>
    Lines Cleared
  </span>
  <span>
    0
  </span>
  <span>
    Level
  </span>
  <span>
    0
  </span>
  <span>
    Single Player Mode
  </span>
</div>
`;

exports[`The Controls Will render the controls for single player mode with an option to switch to multiplayer mode 1`] = `
<div
  className="controls"
>
  <canvas
    height={150}
    tabIndex="0"
    width={210}
  />
  <div
    className="controls__resetPause"
  >
    <PlayControls
      onReset={[MockFunction]}
      onhandlePause={[MockFunction]}
      pauseButtonState={false}
    />
  </div>
  <FontAwesomeIcon
    beat={false}
    beatFade={false}
    border={false}
    bounce={false}
    className="controls__multiplayer"
    fade={false}
    fixedWidth={false}
    flip={false}
    icon={
      Object {
        "icon": Array [
          640,
          512,
          Array [],
          "f0c0",
          "M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z",
        ],
        "iconName": "users",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    maskId={null}
    onClick={[MockFunction]}
    pull={null}
    pulse={false}
    rotation={null}
    shake={false}
    size={null}
    spin={false}
    spinPulse={false}
    spinReverse={false}
    swapOpacity={false}
    symbol={false}
    title=""
    titleId={null}
    transform={null}
  />
  <ForwardRef(Button)
    className="controls__raise"
    onClick={[MockFunction]}
    variant="contained"
  >
    Raise Floor
  </ForwardRef(Button)>
  <span>
    Lines Cleared
  </span>
  <span>
    0
  </span>
  <span>
    Level
  </span>
  <span>
    0
  </span>
  <span>
    Single Player Mode
  </span>
</div>
`;
